From ee56d00a929ed06743d76144728cd77a1c36aa84 Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Wed, 11 Sep 2013 11:33:53 +0800
Subject: [PATCH 60/60] arm: mmp3: enable 10/100M eth by default

10/100M ethernet card is enabled by default in WRLinux, but
it is not very well on hardware, like, software drived LED
light on RJ45 interface. We can disable it by a kernel config
fragment, CONFIG_MACH_QSEVEN_100M_ETH_ENABLE_BY_DEFAULT.

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 arch/arm/mach-mmp/Kconfig |    4 ++++
 arch/arm/mach-mmp/mmp3.c  |    4 ++++
 2 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-mmp/Kconfig b/arch/arm/mach-mmp/Kconfig
index c9dbf92..b894303 100644
--- a/arch/arm/mach-mmp/Kconfig
+++ b/arch/arm/mach-mmp/Kconfig
@@ -98,6 +98,7 @@ config MACH_QSEVEN
 	bool "Marvell's MMP3 QSEVEN Development Board"
 	select CPU_MMP3
 	select HAVE_PWM
+	select MACH_QSEVEN_100M_ETH_ENABLE_BY_DEFAULT
 	select ARM_GIC
 
 endmenu
@@ -146,4 +147,7 @@ config CORE_MORPHING
 	help
 	  Say 'Y' here if you want to enable Marvell MMP core morphing.
 
+config MACH_QSEVEN_100M_ETH_ENABLE_BY_DEFAULT
+	bool
+
 endif
diff --git a/arch/arm/mach-mmp/mmp3.c b/arch/arm/mach-mmp/mmp3.c
index 1dbb6e0..475eb79 100644
--- a/arch/arm/mach-mmp/mmp3.c
+++ b/arch/arm/mach-mmp/mmp3.c
@@ -1763,6 +1763,10 @@ static int __init use_wtm_get_fuse_profile(void)
 	wtm_read_profile();
 	wtm_read_stepping();
 	mmp_plat_ref_clk = wtm_read_platform_config();
+#ifdef CONFIG_MACH_QSEVEN_100M_ETH_ENABLE_BY_DEFAULT
+	/* Temporalily enable 100M ethernet by default */
+	mmp_plat_ref_clk = 0x03;
+#endif
 	wtm_dump_info();
 	return 0;
 }
-- 
1.7.5.4

